dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,115 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+inline void SkipSpace(const char*& ptr) {
+  while(::isspace(*ptr))
+    ptr++;
+}
+
+template <size_t N>
+void Skip(const char*& ptr, const char (&what)[N]) {
```

Comment:
Do we need `ptr` to be a reference to a pointer? It adds a little bit complexity to code, while `sizeof(const char*&)==sizeof(const char*)`

Reply:
The reference `ptr` is there, because we're advancing the pointer.
The reference to array is a the only way to pass a strongly typed sized array (e.g. a string literal) to a function.
`sizeof(what) == N`.
If `what` was declared as `const char what[N]`, the N is meaningless and will be removed, resulting in a type without a size.